crates/ruff_python_parser/src/lexer.rs

```diff
@@ -123,44 +130,73 @@ impl<'src> Lexer<'src> {
         self.current_range
     }
 
+    /// Returns the flags for the current token.
+    pub(crate) fn current_flags(&self) -> TokenFlags {
+        self.current_flags
+    }
+
     /// Helper function to push the given error and return the [`TokenKind::Unknown`] token.
     fn push_error(&mut self, error: LexicalError) -> TokenKind {
         self.errors.push(error);
         TokenKind::Unknown
     }
 
-    /// Lex an identifier. Also used for keywords and string/bytes literals with a prefix.
-    fn lex_identifier(&mut self, first: char) -> TokenKind {
-        // Detect potential string like rb'' b'' f'' u'' r''
-        match (first, self.cursor.first()) {
-            ('f' | 'F', quote @ ('\'' | '"')) => {
-                self.cursor.bump();
-                return self.lex_fstring_start(quote, FStringPrefix::Regular);
+    /// Try lexing the single character string prefix, updating the token flags accordingly.
+    /// Returns `true` if it matches.
+    fn try_single_char_prefix(&mut self, first: char) -> bool {
```

Comment:
Nit: You could consider changing the method to `Option<TokenFlags>` to remove the side effect from it. 

```rust
if let Some(prefix_flags) = self.try_single_char_prefix(first) {
	self.token_flags |= prefix_flags;
  self.lex_string(...)
} else {
	...
}
```

Although it might not be worth it...

Reply:
> Although it might not be worth it...

Yeah, not sure if it's worth doing.

> Nit: I think I would move this method after `lex_identifier`. I was surprised to find this as the very first non-infrastructure method (maybe we should reorganize the lexer methods in a separate PR once we're done with your parser work. E.g. move `next_token` to the top, followed by `lex_token`).

Yes, I'm going to follow-up on this.